### PR TITLE
WINC-1311: [test] Return logs early on success 

### PR DESCRIPTION
**Pull request by mansikulkarni96**

### CODE REVIEW

File: test/e2e/network_test.go [hunk 1]
```diff
@@ -827,31 +828,34 @@ func (tc *testContext) deleteJob(name string) error {
 	return jobsClient.Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
 
-// waitUntilJobSucceeds will return an error if the job fails or reaches a timeout
-func (tc *testContext) waitUntilJobSucceeds(name string) error {
+// waitUntilJobSucceedsAndFetchLogs will return an error if the job fails or reaches a timeout and return logs on success
+func (tc *testContext) waitUntilJobSucceedsAndFetchLogs(name string) (string, error) {
```
Comment (sebsoto) [inline test/e2e/network_test.go:832]: I feel this name change is overly verbose. Its a common pattern for functions like this to return logs, having a note in the function doc as you did should be good enough.

**Review by sebsoto — CHANGES_REQUESTED** [review]

**Review by sebsoto — APPROVED** [review]

**Comment by mansikulkarni96** [comment]
/hold
until vsphere template issue is resolved

**Comment by saifshaikh48** [comment]
/unhold

vsphere issues should be resolved

**Comment by mansikulkarni96** [comment]
/override Red Hat Konflux / windows-machine-config-operator-bundle-master-on-pull-request

**Comment by mansikulkarni96** [comment]
/override Red Hat Konflux / windows-machine-config-operator-bundle-master-on-pull-request

**Comment by mansikulkarni96** [comment]
/override "Red Hat Konflux / windows-machine-config-operator-bundle-master-on-pull-request"

**Pull request merged**